pkg/controller/common/util.go

```diff
@@ -1619,3 +1620,65 @@ func ProgressFromClaim(ctx context.Context, args *ProgressFromClaimArgs) (string
 
 	return "", nil
 }
+
+// IsSnapshotValidForClone returns true if the passed snapshot is valid for cloning
+func IsSnapshotValidForClone(snapshot *snapshotv1.VolumeSnapshot, log logr.Logger) (bool, error) {
```

Comment:
We actually don't care about most of this for planning purposes.  The phase will wait for the snapshot to be ready to use.  I think we only care that the snapshot content exists so we can get the driver.

I don't think the planner has to call this function at all since `ValidateSnapshotCloneProvisioners` validates the driver

Reply:
But this also ensures that the snapshot has a populated status (required for the snapshot content) and it's not errored. I think it doesn't hurt to have it here.

Reply:
At each step in the clone process, we should only do what is absolutely necessary.  There is no benefit to this.  It actually makes things worse because the controller has to do a `requeueAfter` when the planner cannot compute the strategy